src/BUILD

```diff
@@ -591,6 +591,7 @@ genrule(
         "//src/java_tools/buildjar:srcs",
         "//src/java_tools/junitrunner:srcs",
         "//src/java_tools/singlejar:srcs",
+        "//src/tools/singlejar:embedded_tools",
```

Comment:
nit: call this target `:embedded_tools` and make it include the sources of zlib and `src/main/cpp/util`, too.

I was somewhat surprised to see singlejar depending on things under `src/main/cpp`, but here it is:

```
>> bazel query 'somepath(//src/tools/singlejar:singlejar_local,//src/main/cpp/util:path.h)'
//src/tools/singlejar:singlejar_local
//src/tools/singlejar:output_jar
//src/main/cpp/util:util
//src/main/cpp/util:path.h
```

I guess this is not a problem because all the extra dependencies of singlejar are also dependencies of ijar?

Reply:
Which target are you referring to? The sources of zlib and src/main/cpp/util are already included in the zip and including them again is redundant.

Reply:
Absolutely, but it's just a side effect. I.e. if someone removes ijar from the zip, singlejar would break. Not that that's a likely event, but why not do things right?

Reply:
Doing the right thing sounds right :) Done.